documentation/testing/preparations.md

```diff
@@ -0,0 +1,38 @@
+# Preparations
+
+To make testing easier it is a good practice to extract base classes and interfaces and use them as dependencies.
```

Comment:
I had to read this sentence a few times and the code as well to understand what we are talking about. I think it would be worth making it a bit more explicit.
Something like -> "To make testing easier, it is a good practice to inject abstractions (base classes or interfaces) to your classes' constructor, instead of implementations. This will allow us to use fakes or test implementations instead of mocks."